x-pack/plugins/actions/server/lib/is_conector_deprecated.ts

```diff
@@ -18,11 +18,39 @@ export const isConnectorDeprecated = (
      * Connectors after the Elastic ServiceNow application use the
      * Import Set API (https://developer.servicenow.com/dev.do#!/reference/api/rome/rest/c_ImportSetAPI)
      * A ServiceNow connector is considered deprecated if it uses the Table API.
-     *
-     * All other connectors do not have the usesTableApi config property
-     * so the function will always return false for them.
      */
-    return !!connector.config?.usesTableApi;
+
+    /**
+     * We cannot deduct if the connector is
+     * deprecated without config. In this case
+     * we always return false.
+     */
+    if (connector.config == null) {
```

Comment:
a `==` check is dangerous (Javascript shouldn't have ever allowed it IMHO).
Could we replace this with a _positive_ check? that is - ensure `config` is a plain object otherwise return `false`?

Reply:
I wanted to catch `undefined` but you are right. I will do the plain object check. Thanks!

Reply:
Ya, I think we've implicitly typed this stuff internally as `Record<string, unknown>` or such - converting null-ish to empty object on input.  Pretty sure things something would break if you sent a number/string/boolean in as the config.